models/public/fast-neural-style-mosaic-onnx/fast-neural-style-mosaic-onnx.md

```diff
@@ -0,0 +1,110 @@
+# fast-neural-style-mosaic-onnx
+
+## Use Case and High-Level Description
+
+The `fast-neural-style-mosaic-onnx` model is one of the style transfer models
+designed to mix the content of an image with the style of another image. The 
+model uses the method described in [Perceptual Losses for Real-Time Style
+Transfer and Super-Resolution](https://arxiv.org/abs/1603.08155) along with
+[Instance Normalization](https://arxiv.org/pdf/1607.08022.pdf). Original ONNX
+models are provided in the [repository](https://github.com/onnx/models).
+
+
+## Example
+
+
+## Specification
+
+| Metric            | Value            |
+|-------------------|------------------|
+| Type              | Style Transfer   |
+| GFLOPs            | 15.518           |
+| MParams           | 1.679            |
+| Source framework  | PyTorch\*        |
+
+## Accuracy
+
+Accuracy metrics are obtained on MS COCO val2017 dataset. Images were resized to input size.
+
+| Metric | Original model | Converted model (FP32) | Converted model (FP16) |
+| ------ | -------------- | ---------------------- | ---------------------- |
+| PSNR   | 12.03dB        | 12.03dB                | 12.04dB                |
+
+## Performance
+
+## Input
+
+### Original model
+
+Image, name - `input1`,  shape - `1,3,224,224`, format is `B,C,H,W` where:
+
+- `B` - batch size
+- `C` - channel
+- `H` - height
+- `W` - width
+
+Expected color order: RGB.
+
+### Converted model
+
+Image, name - `input1`,  shape - `1,3,224,224`, format is `B,C,H,W` where:
+
+- `B` - batch size
+- `C` - channel
+- `H` - height
+- `W` - width
+
+Expected color order: BGR.
+
+## Output
+
+### Original model
+
+Image, name - `output1`, shape - `1,3,224,224`, format is `B,C,H,W` where:
+
+- `B` - batch size
+- `C` - channel
+- `H` - height
+- `W` - width
+
+Expected color order: RGB.
+
+### Converted model
+
+Image, name - `output1`, shape - `1,3,224,224`, format is `B,C,H,W` where:
+
+- `B` - batch size
+- `C` - channel
+- `H` - height
+- `W` - width
+
+Expected color order: BGR.
```

Comment:
Are you sure about this? Why does the output color order change in the converted model?

Reply:
Because we change color order during conversion in `models/public/fast-neural-style-mosaic-onnx/model.yml`
```
model_optimizer_args:
  - --input_shape=[1,3,224,224]
  - --input_model=$dl_dir/fast-neural-style-mosaic-onnx.onnx
  - --reverse_input_channels
```

Reply:
> 
> 
> Are you sure about this? Why does the output color order change in the converted model?
Because we change color order during conversion in `models/public/fast-neural-style-mosaic-onnx/model.yml`
```
model_optimizer_args:
  - --input_shape=[1,3,224,224]
  - --input_model=$dl_dir/fast-neural-style-mosaic-onnx.onnx
  - --reverse_input_channels
```

Reply:
Yeah, but that is for _input_ channels. I'm asking about the output.

Reply:
Fixed in https://github.com/opencv/open_model_zoo/pull/1106/commits/091a2c0bc82c9475f520169ec5c77bcf4bb8d0fc

Reply:
> 
> 
> Are you sure about this? Why does the output color order change in the converted model?

It is outdated. Expected color order is RGB, fixed in https://github.com/opencv/open_model_zoo/commit/091a2c0bc82c9475f520169ec5c77bcf4bb8d0fc.